sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -157,16 +159,16 @@ object GpuOrcScan {
       meta.willNotWorkOnGpu("GpuOrcScan does not support default values in schema")
     }
 
-    // For date type, timezone needs to be checked also. This is because JVM timezone and UTC
-    // timezone offset is considered when getting [[java.sql.date]] from
-    // [[org.apache.spark.sql.execution.datasources.DaysWritable]] object
-    // which is a subclass of [[org.apache.hadoop.hive.serde2.io.DateWritable]].
+    // When reading timestamp type from an ORC file, it's not related to the
+    // Spark session timezone but the JVM timezone. Currently only supports
+    // UTC and Shanghai timezone.
     val types = schema.map(_.dataType).toSet
-    if (types.exists(GpuOverrides.isOrContainsDateOrTimestamp(_))) {
-      if (!GpuOverrides.isUTCTimezone()) {
-        meta.willNotWorkOnGpu("Only UTC timezone is supported for ORC. " +
-          s"Current timezone settings: (JVM : ${ZoneId.systemDefault()}, " +
-          s"session: ${SQLConf.get.sessionLocalTimeZone}). ")
+    if (types.exists(GpuOverrides.isOrContainsDateOrTimestamp)) {
+      val defaultJvmTimeZone = TimeZone.getDefault
+      if (defaultJvmTimeZone != TimeZone.getTimeZone("UTC")
+        && defaultJvmTimeZone != TimeZone.getTimeZone("Asia/Shanghai")) {
```

Comment:
Only supports UTC and Asia/Shanghai, other timezones like America/New_York need new implementation.